File: leetcode/two-pointers/cpp/283.move-zeroes.cpp
/*
 * @lc app=leetcode id=283 lang=cpp
 *
 * [283] Move Zeroes
 */

// @lc code=start
#include <vector>

using namespace std;

/* class Solution {
public:
	void moveZeroes(vector<int>& nums) {
		int			count = 0;
		int			i = 0;
		
		while (i < nums.size()) {
			if (nums[i] == 0) {
				nums.erase(nums.begin() + i);
				count++;
			}
			else
				i++;
		}
		while(count-- > 0)
			nums.push_back(0);
	}
}; */

class Solution {
public:
	void moveZeroes(vector<int>& nums) {
		size_t	nonzeroCount = 0;
		for (size_t i = 0; i < nums.size(); i++) {
			if (nums[i] != 0)
				nums[nonzeroCount++] = nums[i];
		}
		while (nonzeroCount < nums.size())
			nums[nonzeroCount++] = 0;
	}
};


// @lc code=end

File: leetcode/two-pointers/py/283.move-zeroes.py
#
# @lc app=leetcode id=283 lang=python3
#
# [283] Move Zeroes
#

# @lc code=start
class Solution:
	def moveZeroes(self, nums: List[int]) -> None:
		nonzeroCount = 0;
		for i in range(len(nums)):
			if (nums[i] != 0):
				nums[nonzeroCount] = nums[i]
				nonzeroCount+=1
		while nonzeroCount < len(nums):
			nums[nonzeroCount] = 0;
			nonzeroCount +=1

		
# @lc code=end

